imap_processing/hit/l0/hit_l1a_decom.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from collections import defaultdict
+from enum import IntEnum
+
+import xarray as xr
+
+from imap_processing import decom
+
+logging.basicConfig(level=logging.INFO)
+
+
+class HitAPID(IntEnum):
+    HIT_AUT = 1250  # Autonomy
+    HIT_HSKP = 1251  # Housekeeping
+    HIT_SCIENCE = 1252  # Science
+    HIT_IALRT = 1253  # I-ALiRT
+    HIT_MEMDUMP = 1255  # Memory dump
+
+
+def decom_hit_packets(packet_file: str, xtce: str):
+    """
+    Unpack and decode HIT packets using CCSDS format and XTCE packet definitions.
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+    xtce : str
+        Path to the XTCE packet definition file.
+    Returns
+    -------
+    dict
+        A dictionary containing xr.Dataset for each APID. each dataset in the
+        dictionary will be converted to a CDF.
+    """
+    # TODO: XTCE Files need to be combined
+    logging.info(f"Unpacking {packet_file} using xtce definitions in {xtce}")
+    packets = decom.decom_packets(packet_file, xtce)
+    logging.info(f"{packet_file} unpacked")
+
+    # sort all the packets in the list by their spacecraft time
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+
+    unpacked_data = defaultdict(list)
+    for apid_name, apid in [(id.name, id.value) for id in HitAPID]:
+        # TODO: if science packet, do decompression
+        logging.info(f"Grouping packet values for {apid_name}:{apid}")
+        # get all the packets for this apid and groups them together in a
+        # dictionary
+        unpacked_data[apid_name] = group_apid_data(sorted_packets, apid)
+        logging.info(f"Finished grouping {apid_name}:{apid} packet values")
+
+    # create datasets
+    logging.info("Creating a dataset for HIT L1A data")
+    dataset_dict = create_datasets(unpacked_data)
+    logging.info("HIT L1A dataset created")
+    return dataset_dict
+
+
+def create_datasets(data):
+    """
+    Creates a dataset for each APID in the data
+    Parameters
+    ----------
+    data : dict
+        A single dictionary containing data for all instances of an APID.
+    Returns
+    -------
+    dict
+        A dictionary containing xr.Dataset for each APID. each dataset in the
+        dictionary will be converted to a CDF.
+    """
+    dataset_dict = defaultdict(list)
+    # create one dataset for each APID in the data
+    for apid, data_dict in data.items():
+        # if data for the APID exists, create the dataset
+        if data_dict != {}:
+            epoch = xr.DataArray(
+                name="Epoch", data=data_dict["SHCOARSE"], dims=("Epoch")
```

Comment:
We should get @bryan-harter 's comment on how to name this.

I think this should be `MET` here since it is the mission elapsed time and then we won't confuse it with the reserved epoch name. Then later, I think we need to convert to an EPOCH which I believe needs to be all capitalized, but I also think it has a specific meaning and needs to be offset from a specific date for CDFs.

I don't know the details here though, so hopefully Bryan can give us some guidance.

Reply:
Looks good. Nice job!